### ISSUE TITLE
[Android/IOS] RetroShare for Android/IOS needs Developers

### ISSUE THREAD

**Issue by defnax** [RetroShare/RetroShare#2489]
Hello,
we need more helpers and new devs are welcome to help us

Repository for RetroShare Mobile (Android):
https://github.com/RetroShare/retroshare-mobile

- Todo list for rs mobile:
https://github.com/RetroShare/retroshare-mobile/issues/64

We need helpers maybe port rs mobile for IOS too:
https://flutter.dev/docs/get-started/flutter-for/ios-devs

Howto Setup RetroShare Mobile (Android):
https://retrosharedocs.readthedocs.io/en/latest/developer/retrosharemobile/
https://github.com/RetroShare/retroshare-mobile/blob/master/AndroidStudio-Flutter-setup.md

**Comment by ghost** [comment]
Is there a way to install a recent version of RetroShare Mobile?

**Comment by defnax** [comment]
> Is there a way to install a recent version of RetroShare Mobile?

its not useable yet, after rs mobile goes few minutes to stand by, it loses connection then you need restart everytime.
There is no version to use, not yet usable.

**Comment by selankon** [comment]
Hi, 

We found a solution for this on elrepo.io using a foreground notification. I guess @Kumaravinash9  implemented this:

https://gitlab.com/elRepo.io/elRepo.io-android/-/issues/43

**Comment by defnax** [comment]
> Hi,
> 
> We found a solution for this on elrepo.io using a foreground notification. I guess @Kumaravinash9 implemented this:
> 
> https://gitlab.com/elRepo.io/elRepo.io-android/-/issues/43

yes he implemented it, i tested a build with the foreground but it had same issues, tested on a samsung, rs mobile lost connection after few minutes.
@Kumaravinash9 wanted fix it soon dont know when maybe next year, i hope soon as possible

**Comment by selankon** [comment]
Hi,

We should debug this, could you send me android and miui version? You give foregorund permisions?

Thanks in advance, we didn't detect this bug on elrepo.io yet

**Comment by defnax** [comment]
https://github.com/RetroShare/retroshare-mobile/pull/57


> We should debug this, could you send me android and miui version? You give foregorund permisions?

how to give forground permission? on Android settings?
test his build, seems im only the one tester of the apks
For testing, I has a Samsung Galaxy S8 with Android 9, but i has same issues on a Bluestacks Emulator too.
https://github.com/RetroShare/retroshare-mobile/issues/62

**Comment by selankon** [comment]
> how to give forground permission? on Android settings?

If he followed the guide, a popup asking for permissions should appear on each execution until accepted. If you see the foreground notification probably permissions are set properly. Either you can enable foreground permissions somewhere in the setting I guess... 

We will test this on elrepo.io and I'll give you feedback. 

Thanks!

**Comment by selankon** [comment]
Also, the release apk shouldn't work if not proguard rules added:

https://gitlab.com/elRepo.io/elRepo.io-android/-/issues/50#note_702467138

**Comment by ghost** [comment]
Are there any third party retroshare clients?  Like, clients where you can add friends from retroshare and click on links for chat rooms.
________________________________

**Comment by defnax** [comment]
> > how to give forground permission? on Android settings?
> 
> If he followed the guide, a popup asking for permissions should appear on each execution until accepted. If you see the foreground notification probably permissions are set properly. Either you can enable foreground permissions somewhere in the setting I guess...
> 
> We will test this on elrepo.io and I'll give you feedback.
> 
> Thanks!

thank you i doesnt know this, i doesnt get a popup on Android, not sure when @Kumaravinash9 comes back.
Im only a tester not a Android dev.

**Comment by Kumaravinash9** [comment]
Hi @defnax @selankon . I know how to fix this issues but these days I am busy with my placements and University examination. Once i will free from these . I will fix these issues.  Sorry for the delay.

**Comment by defnax** [comment]
@Kumaravinash9 any news from you?

**Comment by defnax** [comment]
@Kumaravinash9 hi, when you time to continoue on rs mobile/ rs webui?

**Comment by akshaydoshi2** [comment]
Hey guys, is this still active?

**Comment by zelfroster** [comment]
> Hey guys, is this still active?

Yes Akshay, Please wait till someone from the team gets back to you.
cc @csoler @defnax

**Comment by akshaydoshi2** [comment]
> > Hey guys, is this still active?
> 
> Yes Akshay, Please wait till someone from the team gets back to you. cc @csoler @defnax

Okay sure

**Comment by defnax** [comment]
> Hey guys, is this still active?

we need devs for android developement maybe want help to get rs mobile useable?

**Comment by akshaydoshi2** [comment]
> > Hey guys, is this still active?
> 
> we need devs for android developement maybe want help to get rs mobile useable?

Is it flutter or android native? I can work if it's flutter

**Comment by zelfroster** [comment]
> > > Hey guys, is this still active?
> > 
> > 
> > we need devs for android developement maybe want help to get rs mobile useable?
> 
> Is it flutter or android native? I can work if it's flutter

It is indeed Flutter. https://github.com/RetroShare/retroshare-mobile Have a look at it.

**Comment by akshaydoshi2** [comment]
> > > > Hey guys, is this still active?
> > > 
> > > 
> > > we need devs for android developement maybe want help to get rs mobile useable?
> > 
> > 
> > Is it flutter or android native? I can work if it's flutter
> 
> It is indeed Flutter. https://github.com/RetroShare/retroshare-mobile Have a look at it.

Thanks. I did have a look at it.
It needs quite a bit of work.
How does the bounty work here?
Let me know how we should proceed. Would be great if can connect on discord/slack.

**Comment by csoler** [comment]
retroshare-service already works on android. So implementing RS on android is probably very similar to what has been done for the webUI.

**Comment by akshaydoshi2** [comment]
> retroshare-service already works on android. So implementing RS on android is probably very similar to what has been done for the webUI.

Okay sure

**Comment by selankon** [comment]
The actual flutter implementation should download and bundle the RS service automatically. See the instructions and ask here if help needed to run it

**Comment by defnax** [comment]
We use Retroshare for communication, if you want connect with us.

**Comment by akshaydoshi2** [comment]
I cloned the retroshare flutter repository and tried to run the project. It seems that the dependencies amd flutter-sdk need to to upgraded in the project. So I need some time to do that and run the project.

Meanwhile I will download retroshare and connect with you guys over there.

**Comment by defnax** [comment]
i has on my own adroid phone last rs mobile build from @Kumaravinash9 .
which have a big issue
https://github.com/RetroShare/retroshare-mobile/issues/62

**Comment by akshaydoshi2** [comment]
Okay got it. 
Let's connect on Retroshare. There is some issue with installing Retroshare on macOS. So I'll install it tonight on a windows machine.

**Comment by defnax** [comment]
are you on macos only ? i has newer build but at home from 2021, that from gitgub is not useable.
but we has no macos devs, sometime once a year a build macos version for release

i will tonight available too

**Comment by akshaydoshi2** [comment]
macOS is my primary OS. I have windows as well. I'll install retroshare on windows tonight. We can take a look at the macOS release as well if required later on.

**Comment by defnax** [comment]
send me your email i can send my rs key or here but then need to delete the post

**Comment by akshaydoshi2** [comment]
> send me your email i can send my rs key or here but then need to delete the post

akshayd.doshi2@gmail.com

**Comment by defnax** [comment]
mail send

**Comment by akshaydoshi2** [comment]
> mail send

Received. I'll delete my email address from here.

**Comment by defnax** [comment]
@zelfroster can you help to get connected @akshaydoshi2 to the rs network, he has old rs and dht not works, he is on same timezone like you i think

**Comment by zelfroster** [comment]
> @zelfroster can you help to get connected with @akshaydoshi2 to the rs network, he has old rs and dht not works, he is on same timezone like you i think. needs a latest rs with updated dht file

Sure, I can help. You can email me anytime @akshaydoshi2 - sumitkumarsoni123@gmail.com

**Comment by akshaydoshi2** [comment]
> > @zelfroster can you help to get connected with @akshaydoshi2 to the rs network, he has old rs and dht not works, he is on same timezone like you i think. needs a latest rs with updated dht file
> 
> Sure, I can help. You can email me anytime @akshaydoshi2 - [sumitkumarsoni123@gmail.com](mailto:sumitkumarsoni123@gmail.com)

Hey, email sent

**Comment by defnax** [comment]
please delete the post both

**Comment by defnax** [comment]
@akshaydoshi2 i has send you latest windows version, check your mail

**Comment by akshaydoshi2** [comment]
@defnax Yes I have received the email. I will check it tonight

**Comment by defnax** [comment]
hi any news? with rs mobile and with retroshare too? are you connected to dev network?

**Comment by defnax** [comment]
hi any news? we are not connected yet and you doesnt started to check rs mobile? @akshaydoshi2

**Comment by mateli** [comment]
I started https://github.com/RetroShare/RetroShare/issues/2941 "Avalonia + MAUI client (MacOS/Linux/Windows/iOS/Android)" which is a different project but includes an aim to get something running in iOS.

**Comment by defnax** [comment]
> I started [#2941](https://github.com/RetroShare/RetroShare/issues/2941) "Avalonia + MAUI client (MacOS/Linux/Windows/iOS/Android)" which is a different project but includes an aim to get something running in iOS.

Great news

**Comment by defnax** [comment]
@mateli to find how retroshare works, you need join retroshare dev network, there you can meet some devs of rs for feedback, help & questions
